### ISSUE TITLE
Great work! Can I know when the implementation and code of blip-diffusion model be available? 

### ISSUE THREAD

**Issue by WilTay1** [salesforce/LAVIS#361]

**Comment by JPlin** [comment]
+1

**Comment by dxli94** [comment]
Working on this right now. Targeting a release in 2 weeks.

Thanks.

**Comment by ChristiaensBert** [comment]
@dxli94 Awesome! Will it be released on HuggingFace? It's an awesome research

**Comment by dxli94** [comment]
@ChristiaensBert The plan is we will release as part of LAVIS first. HF has kindly agreed in prior for potential integration with diffusers. We will work with HF team to figure that out.

**Comment by ChristiaensBert** [comment]
@dxli94 Nice, I see that you already implemented quite some code 🙌🙌 I wanted to check it out but the checkpoints are not accessible yet 😇 Is there anyway to access and load them?

Also, I would love to help to make an awesome BLIP-Diffusion demo

I've some experience with making extensive Streamlit demo's such as https://huggingface.co/spaces/ml6team/controlnet-interior-design. Let me know if this seems interesting, then we could discuss features to put in the demo :D

**Comment by dxli94** [comment]
@ChristiaensBert we are keen if you could kindly help with 3rd part demo. I'll have to double check with the team and let you know.

**Comment by ChristiaensBert** [comment]
@dxli94 Would be super cool! I could develop the demo in a private space, in a shared organisation, so nothing get published before you guys want. Already started on some basic skeleton app
https://huggingface.co/spaces/BertChristiaens/blip-diffusion

**Comment by ChristiaensBert** [comment]
@dxli94 Any updates?

**Comment by frouhi** [comment]
Any updates on the code release?

**Comment by dxli94** [comment]
Hi @ChristiaensBert @frouhi @JPlin @WilTay1, thanks for the interest and patience. Coming tomorrow.

**Comment by JarvisFei** [comment]
@dxli94 
Amazing results！I am looking forward to it very much.
I saw that you deleted the 20230710-blip-diffusion branch [7 hours ago]. 
When will the code of bilp-diffusion be available?

**Comment by danpaldev** [comment]
> Hi @ChristiaensBert @frouhi @JPlin @WilTay1, thanks for the interest and patience. Please keep an eye on #417.

Hi there, is the implementation not going to be available anymore? The PR appears as closed

**Comment by dxli94** [comment]
Now available on main. Thanks for the patience.

**Comment by danpaldev** [comment]
> Now available on main. Thanks for the patience.

Thanks for letting us know! And of course, thanks for the amazing job.

**Comment by kopyl** [comment]
@dxli94 thank you very much for the released model and notebooks. I've been waiting for this for almost 2 months, now sharing it with my friends.

1: I would suggest adding this line to the notebooks so it's self contained: `!pip install 'git+https://github.com/salesforce/LAVIS.git'` (i will probably make a pull request. But i only checked it with the zero-shot notebook, so going to add it only to that notebook)

2:

you say in the notebook that there is information for fine-tuning a model. But there is no info.


project: [projects/blip-diffusion](https://github.com/salesforce/LAVIS/tree/main/projects/blip-diffusion)
notebook: [link](https://github.com/salesforce/LAVIS/blob/main/projects/blip-diffusion/notebooks/generation_finetuned_dog.ipynb)
the place where you say it: [screenshot](https://i.imgur.com/tMDpu27.png)

Is there any chance you can share the instructions/notebook on the fine-tuning?

3: I was hoping it can give the same results as DreamBooth but without the actual training.  But the results are not the best:

the input image and some generated images: https://imgur.com/a/tfGB2ip
the prompt: `wearing hat`

So the question: why this thing is needed when we have the DreamBooth which requires training while Blip Diffusion also requires training to give good results in the accuracy similar to DreamBooth (I did not test it myself though)

**Comment by dxli94** [comment]
Hello @kopyl,

Thanks for your questions and PR. I've merged PR. Regarding the questions:

2. unfortunately, we are not able to release the fine-tuning code due to internal concerns. However, I'd be happy to receive concrete questions on fine-tuning. 

3.  BLIP-Diffusion fine-tunes 10-25x times faster than DreamBooth. This means you can get one personalized model in 30 seconds, rather 5-10 minutes as in DreamBooth, or 20-30 minutes as in textual inversion. This is critical for scalable applications or those are time-sensitive, for example, in multimodal dialogue systems. 

In the paper, we report quantitative evaluations using both zero-shot and fine-tuning setups. The current version of BLIP-Diffusion achieves comparable or better visual quality compared to DreamBooth. The zero-shot personalization capability is new, and as far as we know, not available in other models on the market. However, do notice the gap in evaluation between the zero-shot and fine-tuning results.

Can you better elaborate are the images here generated from BLIP-Diffusion https://imgur.com/a/tfGB2ip?

**Comment by kopyl** [comment]
@dxli94 
1. thank you. I reached out to you on LinkedIn.
2. Yeah, thanks. Just read the paper in mode details right after I made this comment. Sorry, I should have done it before asking. This is a really good point actually, but such dumb devs as me can't implement the fine-tuning based on just paper :(

I actually noticed that the image quality seems to be better than Stable Diffusion for some reason.

4. Did you train the base model from scratch?
5. How did you train it?
6. How much time did it take to train? (so I could calculate the cost. or would be even better if you tell the cost)

> In the future, we strive to minimize this gap
Amazing!!! Can't wait :)

Yeah, those images were generated zero-shot with BLIP-Diffusion. First one is the input image, others – generated.

**Comment by arjun810** [comment]
When using the example fine-tuned checkpoint, it appears that it's heavily overtrained. It is difficult to get an output in other styles that work well on regular stable diffusion 1.5. Any tips for dealing with that?

**Comment by arjun810** [comment]
Also, I know finetuning isn't available publicly, but if we were to try to write it ourselves, how much GPU memory would it require?

**Comment by dxli94** [comment]
Editing capabilities are now released.

See https://github.com/salesforce/LAVIS/commit/f982acc73288408bceda2d35471a8fcf55aa04ca.

Thanks.

**Comment by kopyl** [comment]
@dxli94 the [tryon](https://github.com/salesforce/LAVIS/blob/main/projects/blip-diffusion/notebooks/editing_tryon_zeroshot.ipynb) one actually looks nice.

But the dress does not really look like in the picture.

Do you think the results might be way better with training? And what kind of images in a dataset should be used to have the best resemblance?

**Comment by dxli94** [comment]
@kopyl the try on demo is without fine-tuning. With fine-tuning, results can be improved. For this particular application, we find it sufficient to fine-tune on one single image of cloth.

**Comment by kopyl** [comment]
@dxli94 could you please share generated images with the model fine-tuned on the clothing? + photo of the clothing used for the fine-tuning :)
